### ISSUE TITLE
Support for Proxy server

### ISSUE THREAD

**Issue by reiabreu** [Screenly/Anthias#158]
Screenly OSE  should support proxy servers and proxy exceptions

**Comment by fridomac** [comment]
Sorry to dig up this old issue, but we need to use a proxy for the RPi´s...
Tried the proxy support, but when I switch to the "158" branch, I only get a black screen. Web GUI responds normally, but I can not add Webpages ("Server can not retrieve Asset"). 
Is there any log I can look into?
Bye, frido.

**Comment by vpetersson** [comment]
That branch is pretty much out of date. What we need to do is to pull in and merge the latest changes, and then issue a pull request. Unfortunately I don't have enough time to do that right now.

**Comment by fridomac** [comment]
No problem, I will try and set the proxy for uzbl manually, see how far I´ll come (i presume I have to set it in ./config somewhere, or in misc/uzbl.rc?

**Comment by vpetersson** [comment]
@over64 Let's just porting [this](https://github.com/wireload/screenly-ose/commit/7c577e56bf9e5ff1661417db4a8928c3d46e279f) should be enough.

**Comment by varazir** [comment]
Is there any official support for a proxy?

**Comment by vpetersson** [comment]
@varazir No, that's the point of this issue is to add that.

**Comment by jamheadjames64** [comment]
Question since this has been removed from sprint2, how do i get the latest updates that add the misc folder to the image?

Sorry I'm just not that great with github, thanks.

**Comment by vpetersson** [comment]
Depends on your setup, but since you're commenting on this, I presume you require a proxy, you might try [this](https://www.raspberrypi.org/forums/viewtopic.php?f=28&t=18634 and then follow the regular installation instructions.

**Comment by jamheadjames64** [comment]
Have done, the Pi works to apt-get update but within Screenly it doesn't. When I try to put in a asset for website is cant with "Server Error: Could not retrieve file. Check the asset URL" for http://www.bbc.co.uk/news

**Comment by vpetersson** [comment]
Yeah this is because the proxy support in the system is implemented, but not in the actual Screenly stack.

**Comment by tyronewyatt** [comment]
I use redsocks on my screenly ose and point it to our internal proxy server. 
Here is a basic guide to get you started https://jmkhael.io/escape-proxy-hell-with-redsocks/

**Comment by vpetersson** [comment]
Interesting. Good find @tyronewyatt! Thanks for sharing.
